dbms/src/Common/tests/gtest_mytime.cpp

```diff
@@ -121,6 +121,57 @@ try
     {
         checkParseMyDateTime(str, expected, type_with_fraction);
     }
+    cases_with_fsp = {
+        {"2012-12-31 11:30:45", "2012-12-31 11:30:45"},
+        {"0000-00-00 00:00:00", "0000-00-00 00:00:00"},
+        {"0001-01-01 00:00:00", "0001-01-01 00:00:00"},
+        {"00-12-31 11:30:45", "2000-12-31 11:30:45"},
+        {"12-12-31 11:30:45", "2012-12-31 11:30:45"},
+        {"2012-12-31", "2012-12-31 00:00:00"},
+        {"20121231", "2012-12-31 00:00:00"},
+        {"121231", "2012-12-31 00:00:00"},
+        {"2012^12^31 11+30+45", "2012-12-31 11:30:45"},
+        {"2012^12^31T11+30+45", "2012-12-31 11:30:45"},
+        {"2012-2-1 11:30:45", "2012-02-01 11:30:45"},
+        {"12-2-1 11:30:45", "2012-02-01 11:30:45"},
+        {"20121231113045", "2012-12-31 11:30:45"},
+        {"121231113045", "2012-12-31 11:30:45"},
+        {"2012-02-29", "2012-02-29 00:00:00"},
+        {"00-00-00", "0000-00-00 00:00:00"},
+        // {"00-00-00 00:00:00.123", "2000-00-00 00:00:00.123"},
```

Comment:
Test cases use fixed `fsp`; automatically calculating the `fsp` of a String is not implemented yet. I think we can leave these cases for later.